Library/Formula/stunnel.rb

```diff
@@ -65,9 +66,35 @@ def caveats
         #{etc}/stunnel/stunnel.pem
 
       This certificate will be used by default unless a config file says otherwise!
+      Stunnel will refuse to load the sample configuration file if left unedited.
 
       In your stunnel configuration, specify a SSL certificate with
       the "cert =" option for each service.
     EOS
   end
+
+  test do
+    ENV["HOME"] = testpath
+    (testpath/"tstunnel.conf").write <<-EOS.undent
+      cert = #{etc}/stunnel/stunnel.pem
+
+      setuid = nobody
+      setgid = nobody
+
+      [pop3s]
+      accept  = 995
+      connect = 110
+
+      [imaps]
+      accept  = 993
+      connect = 143
+    EOS
+
+    # Stunnel throws non-zero exit codes on *ANY* failure like a spoilt child, hence
+    # we need to stupidly guard this. Although this is nigh-useless for the bot, the test
+    # allows maintainers and users to manually verify stunnel is behaving as expected.
+    # If words "Configuration successful" appear in `brew test --verbose` then test = success.
+    # The exit 1 error is because to bind ports already in use you tend to require `sudo`.
+    system "(#{bin}/stunnel #{testpath}/tstunnel.conf; true)"
```

Comment:
Doesn't seem to work as hoped :/.

```
    output = shell_output("#{bin}/stunnel #{testpath}/tstunnel.conf; true")
    assert output.include?("Configuration successful")
```

```
Error: stunnel: failed
Failed assertion, no message given.
Kernel.exit
```

Same failure for ` assert pipe_output("#{bin}/stunnel #{testpath}/tstunnel.conf; true").include?("successful") `.

It's definitely there though, in the output:

```
[ ] SSL options: 0x03000004 (+0x03000000, -0x00000000)
[.] Configuration successful
[ ] Listening file descriptor created (FD=6)
[!] Error binding service [pop3s] to 0.0.0.0:995
```